frontend/src/lib/components/EditableText.svelte

```diff
@@ -0,0 +1,157 @@
+﻿<script lang="ts">
+	import { LoadingIcon } from '$lib/icons';
+	import { ZodString, z } from 'zod';
+	import { _changeProjectName } from '../../routes/project/[project_code]/+page';
+	import IconButton from './IconButton.svelte';
+	import { Form, lexSuperForm, lexSuperValidate } from '$lib/forms';
+
+	export let value: string | undefined | null = undefined;
+	export let disabled = false;
+	export let saveHandler: (newValue: string) => Promise<unknown>;
+	export let placeholder: string | undefined = undefined;
+	export let multiline = false;
+	export let validation: ZodString | undefined = undefined;
+
+	let initialValue: string | undefined | null;
+	let editing = false;
+	let saving = false;
+	const fieldId = 'id' + new Date().getTime();
+
+	let inputElem: HTMLInputElement | HTMLTextAreaElement;
+
+	const formSchema = z.object(validation ? { [fieldId]: validation } : {});
+	let { form, errors, valid, update, reset } = lexSuperForm(
+		formSchema,
+		{ taintedMessage: false },
+	);
+	$: error = $errors[fieldId]?.join(', ');
+
+	function startEditing() {
+		if (disabled) {
+			return;
+		}
+
+		initialValue = value;
+		form.set({[fieldId]: value ?? ''});
+		editing = true;
+	}
+
+	async function save() {
+		const newValue = inputElem.value;
```

Comment:
we should avoid using elements directly wherever possible. Can we get the value from `$form` here? If it's not updated at this point, then you can pass it in from the new callback in `lexSuperForm`

Reply:
Looks like you fixed it 🎉